src/Stripe/Controllers/PaymentIntentRequest.php

```diff
@@ -34,14 +33,19 @@ public function __invoke()
             ]);
         }
 
+        // 🏴‍☠️ Ensure no reCaptcha fails
+        if ($data->enableRecaptcha) {
```

Comment:
@DevinWalker Looking at where `$data` is coming from, whether or not reCAPTCHA is enabled shouldn't be sent from the front-end. Otherwise, someone can simply say that it's disabled and spam the endpoint again. Rather, this should be determined purely by the server based on the block.

Reply:
Oh boy, good point. This is going to be a bit of a doozy to refactor. 

I'm thinking its easiest to use a `wp_option` to flag whether ReCAPTCHA is enabled then use that option for all forms on the site. I can't think of a good reason why an admin would want to enable ReCAPTCHA on one form vs another. If you're getting attacked by bots, it's best to enable it for all forms.

If we wanted to enable ReCAPTCHA conditionally based on the block setting then it would require more in depth thinking about how to detect that option per block. Gutenberg doesn't make this easy for us and I spoke with Ben Ritner and he pointed me to how Kadence forms does it and it's not pretty: https://github.com/kadencewp/kadence-blocks/blob/7b49e80eef72c2d0df7e1a9a335b81d332f6a8bc/includes/form-ajax.php#L673

I made some progress in another branch on the wp_option route. Looking forward to chatting more about how to best tackle this next week.

Reply:
Resolved in latest push